app/code/Magento/GoogleAnalytics/Helper/Data.php

```diff
@@ -33,19 +49,77 @@ class Data extends \Magento\Framework\App\Helper\AbstractHelper
      */
     public function isGoogleAnalyticsAvailable($store = null)
     {
-        $accountId = $this->scopeConfig->getValue(self::XML_PATH_ACCOUNT, ScopeInterface::SCOPE_STORE, $store);
-        return $accountId && $this->scopeConfig->isSetFlag(self::XML_PATH_ACTIVE, ScopeInterface::SCOPE_STORE, $store);
+        return $this->scopeConfig->isSetFlag(self::XML_PATH_ACTIVE, ScopeInterface::SCOPE_STORE, $store) && $this->getAccountId();
     }
 
     /**
      * Whether anonymized IPs are active
-     *
+     * Google Analytics Accounts(GA4) is always true
      * @param null|string|bool|int|Store $store
      * @return bool
      * @since 100.2.0
      */
     public function isAnonymizedIpActive($store = null)
     {
-        return (bool)$this->scopeConfig->getValue(self::XML_PATH_ANONYMIZE, ScopeInterface::SCOPE_STORE, $store);
+        if ($this->isGoogleAnalytics4Account()) {
+            return true;
+        } else {
+            return (bool)$this->scopeConfig->getValue(self::XML_PATH_ANONYMIZE, ScopeInterface::SCOPE_STORE, $store);
+        }
+
+    }
+
+    /**
+     * Checks if Account Type is Google Analytics Account
+     *
+     * @return bool
+     */
+    public function isGoogleAnalytics4Account()
+    {
+        return $this->scopeConfig->getValue(
+            self::XML_PATH_ACCOUNT_TYPE, \Magento\Store\Model\ScopeInterface::SCOPE_STORE) == self::ACCOUNT_TYPE_GOOGLE_ANALYTICS4;
+    }
+
+    /**
+     * Checks if Account Type is Universal Analytics Account
+     *
+     * @return bool
+     */
+    public function isUniversalAnalyticsAccount()
+    {
+        return $this->scopeConfig->getValue(
+            self::XML_PATH_ACCOUNT_TYPE,\Magento\Store\Model\ScopeInterface::SCOPE_STORE) == self::ACCOUNT_TYPE_UNIVERSAL_ANALYTICS;
+    }
+
+    /**
+     * Get Account Id, depending on property type Tracking Id (UA) or Measurement Id (GA4)
+     *
+     * @return string
+     */
+    public function getAccountId()
+    {
+        if ($this->isGoogleAnalytics4Account()) {
```

Comment:
else is not required here. You can go with just `if` or consider a ternary operator.